source/game/console.py

```diff
@@ -0,0 +1,142 @@
+""" Plays the game of Colonies """
+from itertools import cycle
+from source.game.logger import logger
+from sys import exit
+
+
+class Console:
+
+    def __init__(self, board):
+        """
+        Constructor for a console.
+        :param board: Provided game board object.
+        :type board: Board
+        """
+        self.board = board
+        self.active_player = None
+
+    def evaluate_board(self):
+        """ Handles any updates on board state. Exits if there is a winner. """
+        if self.is_winner() is True:
+            logger.error("Player=%u has won!.", self.active_player.id)
+            exit()
+
+        self.board.display_text_board()
+
+    def get_next_player(self):
+        """ Switches to the next avaiable player with pieces."""
+        if self.active_player is None:
+            self.active_player = self.board.players[0]
+        else:
+            players_cycle = cycle(self.board.players)
+
+            # Cycle to the current player.
+            for temp_player in players_cycle:
+                if temp_player is self.active_player:
+                    break
+
+            # Cycle to the next available player with pieces.
+            for temp_player in players_cycle:
+                if (self.board.does_player_have_pieces(temp_player) is False):
+                    continue
+                self.active_player = temp_player
+                break
+
+        logger.debug("Starting turn for player=%u.", self.active_player.id)
+
+    def take_turn(self):

```

Comment:
The `take_turn` function is pretty long. Though not required, for clarity perhaps some of its parts should be extracted into separate functions.